android/library/src/test/java/io/logto/android/auth/browser/BrowserSignInFlowTest.kt

```diff
@@ -153,93 +167,119 @@ class BrowserSignInFlowTest {
 
         browserSignInFlow.handleRedirectUri(invalidUri)
 
-        verify(onCompleteMock).invoke(logtoExceptionCaptor.capture(), tokenSetCaptor.capture())
-        assertThat(logtoExceptionCaptor.firstValue)
+        verify { onCompleteMock.invoke(capture(logtoExceptionSlot), isNull()) }
+        assertThat(logtoExceptionSlot.captured)
             .hasMessageThat()
             .isEqualTo("$SIGN_IN_FAILED: $INVALID_REDIRECT_URI")
-        assertThat(tokenSetCaptor.firstValue).isNull()
+    }
+
+    @Test
+    fun handleRedirectUriShouldCompleteWithLogtoExceptionWithUriWithoutState() {
```

Comment:
what about invalid state?

Reply:
TBD: or unknown state? @simeng-li

Reply:
in this test case there is no state passed in the redirectUri.  Need to test an unmatched state being returned.